### PR TITLE
little fixes (including applying formatting

### LINKED ISSUE #27: simplify predicate
https://github.com/OverkillGuy/qrxfil/blob/58d160ff019b40b66899a125b015e6f6dd6409e4/src/pdf.rs#L29

i think you can probably change this to

```rust
.filter(|file| file.ends_with("png")) 
```
or 
```rust
.filter(|file| file.extension() == Some("png")) 
```

something along those lines

### PR DESCRIPTION
**Pull request by danieleades**
closes #27

### CODE REVIEW

File: tests/integration_tests.rs [hunk 1]
```diff
@@ -15,11 +15,6 @@
 // along with this program. If not, see
 // <https://www.gnu.org/licenses/>.
 
-extern crate assert_cmd;
-extern crate assert_fs;
-extern crate predicates;
-extern crate rand;
-
```
Comment (OverkillGuy) [inline tests/integration_tests.rs:22]: 100% glad to see this go. I thought I had removed them already, but apparently haven't.

File: src/pdf.rs [hunk 2]
```diff
@@ -17,21 +17,21 @@
 
 //! Create PDFs for printed paper backup
 
-use pandoc;
```
Comment (OverkillGuy) [inline src/pdf.rs:20]: How can this work? Below (line 72) I use `pandoc::new()`, how can that work without importing pandoc in this file?

File: src/pdf.rs [hunk 3]
```diff
@@ -17,21 +17,21 @@
...
-        .filter(|file| file.file_name().to_str().unwrap().ends_with("png"))
-        .map(|e| e.path())
-        .collect();
-    // read_dir does not guarantee ordering => explicit sort chunk files
-    output_files.sort();
-    output_files
+        .map(|file| file.path())
+        .filter(|path| path.ends_with("png"))
+        // read_dir does not guarantee ordering => explicit sort chunk files
+        .sorted()
```
Comment (OverkillGuy) [inline src/pdf.rs:33]: Awww I thought I looked for this but didn't find it! I wanted the equivalent for python's `sorted(mylist)` to make it inline, but only found `list.sort()` which sucked. Good find, thanks!

File: src/csv_decode.rs [hunk 4]
```diff
@@ -15,9 +15,9 @@
 // along with this program. If not, see
 // <https://www.gnu.org/licenses/>.
 
-//! CSV decoder to read BinaryEye app's "CSV with semicolon delimiter"
+//! CSV decoder to read `BinaryEye` app's "CSV with semicolon delimiter"
 //! export format. Inspired by
-//! https://docs.rs/csv/1.1.6/csv/#example-with-serde
+//! <https://docs.rs/csv/1.1.6/csv/#example-with-serde>
```
Comment (OverkillGuy) [inline src/csv_decode.rs:20]: Am assuming those were manual edits. Were you doing this checking `cargo doc` every time, or was there a trick tool I don't know about?

File: src/parser.rs [hunk 5]
```diff
@@ -123,10 +123,9 @@ fn collapse_chunks(chunks: &[EncodedChunk]) -> Result<Vec<EncodedChunk>, Restore
     let mut mismatching = Vec::<(EncodedChunk, String)>::new();
     // Group by chunk id
     for (_chunk_id, mut duplicate_chunks) in &chunks.iter().group_by(|chunk| chunk.id) {
-        let reference_chunk = match duplicate_chunks.next() {
-            Some(v) => v,
-            None => panic!("Couldn't get 1 item per groupby"),
-        };
+        let reference_chunk = duplicate_chunks
+            .next()
+            .expect("Couldn't get 1 item per groupby");
```
Comment (OverkillGuy) [inline src/parser.rs:128]: Oh yeah I forgot that this was a thing...
Good practice to have this kind of code smell debunked, thanks!

**Review by OverkillGuy — APPROVED** [review]
Super happy to take those in 100%, just curious about some of these (see inline)

**Review by OverkillGuy — CHANGES_REQUESTED** [review]
Haaaaang on. I just ran the `pdfprint` subcommand and found a bug:  `read_folder_sorted` returns 0 items now. Investigating...

**Comment by OverkillGuy** [comment]
@danieleades Check out [pdf-test branch](https://github.com/OverkillGuy/qrxfil/tree/pdf-test) to see the pdf-lacking-chunks pop up.

File: src/pdf.rs (hunk 2 above)
Comment (danieleades) [reply src/pdf.rs:20]: external crates are implicitly imported (its the `extern crate <crate>` invocation that's implicit) in 2018 edition

File: src/pdf.rs (hunk 3 above)
Comment (danieleades) [reply src/pdf.rs:33]: well, i figured if you're using `itertools`, you might as well use `itertools`...

File: src/csv_decode.rs (hunk 4 above)
Comment (danieleades) [reply src/csv_decode.rs:20]: `cargo clippy` showed this as a warning

**Comment by danieleades** [comment]
> Haaaaang on. I just ran the `pdfprint` subcommand and found a bug: `read_folder_sorted` returns 0 items now. Investigating...
> 
> EDIT: Will be adding a test for this issue (PDF _was_ generated but had only the cover page!)

is it working in master?

**Comment by OverkillGuy** [comment]
> is it working in master?

Yessir!

**Comment by OverkillGuy** [comment]
Found the issue:
> ends_with()
> Determines whether child is a suffix of self.
> Only considers whole path components to match.
> 
> Examples
> ```rust
> use std::path::Path;
> 
> let path = Path::new("/etc/resolv.conf");
> 
> assert!(path.ends_with("resolv.conf"));
> assert!(path.ends_with("etc/resolv.conf"));
> assert!(path.ends_with("/etc/resolv.conf"));
> 
> assert!(!path.ends_with("/resolv.conf"));
> assert!(!path.ends_with("conf")); // use .extension() instead
> ```

**Comment by danieleades** [comment]
> Found the issue:
> 
> > ends_with()
> > Determines whether child is a suffix of self.
> > Only considers whole path components to match.
> > Examples
> > ```rust
> > use std::path::Path;
> > 
> > let path = Path::new("/etc/resolv.conf");
> > 
> > assert!(path.ends_with("resolv.conf"));
> > assert!(path.ends_with("etc/resolv.conf"));
> > assert!(path.ends_with("/etc/resolv.conf"));
> > 
> > assert!(!path.ends_with("/resolv.conf"));
> > assert!(!path.ends_with("conf")); // use .extension() instead
> > ```

try this...

**Comment by OverkillGuy** [comment]
> try this...

Yup, fixes the issue. Will take that in

**Comment by OverkillGuy** [comment]
Adopted this in 277dff8, closing. Apparently this commit lost the attribution, but it's all thanks to you!

**Pull request closed without merging**